### PR TITLE
Organize tutorials sections

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR continues some of the improvements to the organization of our tutorials proposed in #4077, but adding an extra level of organization.

I have gotten something wrong though, as I don't see the extra level and what I thought would be a link is not a link

image

I've also gotten some warnings

```
/Users/nsofroniew/GitHub/napari/docs/howtos/layers/points.md:28: WARNING: 'myst' reference target not found: ../../tutorials/applications/annotate_points
/Users/nsofroniew/GitHub/napari/docs/howtos/layers/points.md:92: WARNING: 'myst' reference target not found: ../../tutorials/applications/annotate_points
/Users/nsofroniew/GitHub/napari/docs/tutorials/annotation/index.md:5: WARNING: 'myst' reference target not found: annotate_points
/Users/nsofroniew/GitHub/napari/docs/tutorials/processing/index.md:5: WARNING: 'myst' reference target not found: dask
/Users/nsofroniew/GitHub/napari/docs/tutorials/segmentation/index.md:5: WARNING: 'myst' reference target not found: annotate_segmentation
/Users/nsofroniew/GitHub/napari/docs/tutorials/tracking/index.md:5: WARNING: 'myst' reference target not found: cell_tracking
```

So clearly there is a bit more work to do before this is done, but if anyone had any tips on how to fix the nesting & links that would be great, thanks!

### CODE REVIEW

**Comment by melissawm** [comment]
Looks great! One thing that might be helpful is open documentation issues with content requests for each of those sections, so that people know how to contribute.

**Comment by sofroniewn** [comment]
> Looks great! One thing that might be helpful is open documentation issues with content requests for each of those sections, so that people know how to contribute.

Nice I like that idea.

Also @melissawm I'm running into the following myst reference target not found errors, but i'm not sure why - are they typos? the wrong link formatting?

```
/Users/nsofroniew/GitHub/napari/docs/howtos/layers/points.md:28: WARNING: 'myst' reference target not found: ../../tutorials/annotation/annotate_points
/Users/nsofroniew/GitHub/napari/docs/howtos/layers/points.md:92: WARNING: 'myst' reference target not found: ../../tutorials/annotation/annotate_points
/Users/nsofroniew/GitHub/napari/docs/tutorials/annotation/index.md:5: WARNING: 'myst' reference target not found: annotate_points
/Users/nsofroniew/GitHub/napari/docs/tutorials/processing/index.md:5: WARNING: 'myst' reference target not found: dask
/Users/nsofroniew/GitHub/napari/docs/tutorials/segmentation/index.md:5: WARNING: 'myst' reference target not found: annotate_segmentation
/Users/nsofroniew/GitHub/napari/docs/tutorials/tracking/index.md:5: WARNING: 'myst' reference target not found: cell_tracking
```

The things aren't links too. Do you have any idea how to fix? Maybe @jni @Carreau or @tlambert03 knows too - sorry if this is super simple thing, i just can't seem to figure it out!!

**Comment by melissawm** [comment]
This is because the default for the current ToC setup is to ignore files that are not included in the ToC - so links to them aren't generated. I'll note the fix below.

File: docs/_toc.yml [hunk 1]
```diff
@@ -16,12 +16,10 @@ subtrees:
       - file: tutorials/index
         subtrees:
         - entries:
+          - file: tutorials/annotation/index
+          - file: tutorials/processing/index
           - file: tutorials/segmentation/index
           - file: tutorials/tracking/index
```
Comment (melissawm) [inline docs/_toc.yml]:
```suggestion
          - file: tutorials/annotation/index
            subtrees:
            - entries: 
              - file: tutorials/annotation/annotate_points
          - file: tutorials/processing/index
            subtrees:
            - entries:
              - file: tutorials/processing/dask
          - file: tutorials/segmentation/index
            subtrees:
            - entries:
              - file: tutorials/segmentation/annotate_segmentation
          - file: tutorials/tracking/index
            subtrees:
            - entries:
              - file: tutorials/tracking/cell_tracking
```

Comment (melissawm) [reply docs/_toc.yml]: The last error should be fixed in L98 of the points how-to: the link to the point annotation tutorial should be
```
[point annotation tutorial](../../tutorials/annotation/annotate_points)
```

**Comment by sofroniewn** [comment]
> This is because the default for the current ToC setup is to ignore files that are not included in the ToC - so links to them aren't generated. I'll note the fix below.

Ah ok great - that makes sense. Maybe worth capturing in our docs contributing guide - cc @DragaDoncila

File: docs/_toc.yml (hunk 1 above)
Comment (sofroniewn) [reply docs/_toc.yml]: got it, thanks!

**Comment by sofroniewn** [comment]
This is now working great - thanks for help @melissawm. PR is ready for review

image

**Review by jni — APPROVED** [review]

**Pull request merged**